src/ApiPlatform/Resources/CartRule.php

```diff
@@ -29,16 +29,14 @@
 namespace PrestaShop\Module\APIResources\ApiPlatform\Resources;
 
 use ApiPlatform\Metadata\ApiResource;
-use ApiPlatform\Metadata\Put;
 use PrestaShop\PrestaShop\Core\Domain\CartRule\Command\EditCartRuleCommand;
-use PrestaShopBundle\ApiPlatform\Processor\CommandProcessor;
+use PrestaShopBundle\ApiPlatform\Metadata\CQRSPartialUpdate;
 
 #[ApiResource(
     operations: [
-        new Put(
+        new CQRSPartialUpdate(
```

Comment:
If it as the `PUT` method then we should use `CQRSUpdate`
`CQRSPartialUpdate` is used for `PATH` method